nni/mutable/shortcut.py

```diff
@@ -0,0 +1,235 @@
+                        'Use `.int()` to convert to integer.', quantize)
+    if low >= high:
+        raise ValueError('low must be strictly smaller than high.')
+
+    return Numerical(low, high, quantize=quantize, label=label)
+
+
+def loguniform(label: str, low: float, high: float) -> Numerical:
+    """Draw from a range [low, high] according to a loguniform distribution::
+
+        exp(uniform(log(low), log(high))),
+
+    so that the logarithm of the return value is uniformly distributed.
+
+    Since logarithm is taken here, low and high must be strictly greater than 0.
+
+    This is often used in variables which are log-distributed in experience,
+    such as learning rate (which we often choose from 1e-1, 1e-3, 1e-6...).
+
+    Examples
+    --------
+    >>> nni.loguniform('x', 1e-5, 1e-3)
+    Numerical(1e-05, 0.001, log_distributed=True, label='x')
+    >>> list(nni.loguniform('x', 1e-5, 1e-3).grid(granularity=2))
+    [3.1622776601683795e-05, 0.0001, 0.00031622776601683794]
+    """
+    if low >= high:
+        raise ValueError('low must be strictly smaller than high.')
+    if low <= 0 or high <= 0:
+        raise ValueError('low and high must be strictly greater than 0.')
+    return Numerical(low, high, log_distributed=True, label=label)
+
+
+def qloguniform(label: str, low: float, high: float, quantize: float) -> Numerical:
+    """A combination of :func:`quniform` and :func:`loguniform`.
+
+    Note that the quantize is done **after** the sample is drawn from the log-uniform distribution.
+
+    Examples
+    --------
+    >>> nni.qloguniform('x', 1e-5, 1e-3, 1e-4)
+    Numerical(1e-05, 0.001, q=0.0001, log_distributed=True, label='x')
+    """
+    return Numerical(low, high, log_distributed=True, quantize=quantize, label=label)
+
+
+def normal(label: str, mu: float, sigma: float) -> Numerical:
+    """Declare a normal distribution with mean ``mu`` and standard deviation ``sigma``.
+
+    The variable is unbounded, meaning that any real number from ``-inf`` to ``+inf`` can be possibly sampled.
+
+    Examples
+    --------
+    >>> nni.normal('x', 0, 1)
+    Numerical(-inf, inf, mu=0, sigma=1, label='x')
+    >>> nni.normal('x', 0, 1).random()
+    -0.30621273862239057
+    """
+    if sigma <= 0:
+        raise ValueError('Standard deviation must be strictly greater than 0.')
+
+    return Numerical(mu=mu, sigma=sigma, label=label)
+
+
+def qnormal(label: str, mu: float, sigma: float, quantize: float) -> Numerical:
+    """Similar to :func:`quniform`, except the uniform distribution is replaced with a normal distribution.
+
+    Examples
+    --------
+    >>> nni.qnormal('x', 0., 1., 0.1)
+    Numerical(-inf, inf, mu=0.0, sigma=1.0, q=0.1, label='x')
+    >>> nni.qnormal('x', 0., 1., 0.1).random()
+    -0.1
+    """
+    return Numerical(mu=mu, sigma=sigma, quantize=quantize, label=label)
+
+
+def lognormal(label: str, mu: float, sigma: float) -> Numerical:
```

Comment:
I suggest to remove `lognormal` and `qlognormal`. They are misleading and really unnecessary.

Reply:
Agreed. Let's make sure @QuanluZhang agrees with us.

Reply:
we can follow our discussion.